src/StatsigOnPrem.ts

```diff
@@ -150,30 +176,60 @@ export default class StatsigOnPrem implements StatsigInterface {
       name,
       salt: IDUtils.generateNewSalt(),
       idType: idType ?? "userID",
+      targetApps: new Set(targetApps ?? []),
       ...metadata,
     };
-    await this.store.addGate(name, gate, targetApps);
-    await this.cache.clearAll();
+    await this.store.addGate(gate);
+    await this.clearCacheForEntity(gate);
   }
 
   public async updateGate(
     name: string,
     args: Partial<FeatureGateCreationArgs>
   ): Promise<void> {
-    const { targetApps, ...metadata } = args;
-    const existing = await this.getGate(name);
-    if (existing == null) {
+    const gate = await this.getGate(name);
+    if (gate == null) {
       console.warn("Attempting to update non-existent gate");
       return;
     }
-    const updatedGate: FeatureGate = { ...existing, ...metadata };
-    await this.store.updateGate(name, updatedGate, targetApps);
-    await this.cache.clearAll();
+    await this.store.updateGate(gate, args);
```

Comment:
`gate` seems to be the previous state, should this be passing in `updatedGate` instead?

Reply:
I revised the logic a bit here to pass in the original gate along with the updated fields. Just to keep the interface logic simple. The gate is updated in-place

Reply:
Ah right, I missed that 🤦 

Looks like `updateGate` makes its own copy [here](https://github.com/statsig-io/statsig-on-prem/blob/d51a8e5ba10adb067e17d09e37492ae9d569d5ae/src/utils/StorageHandler.ts#L40), I suspect that means the `gate` that is passed to `clearCacheForEntity` is still the old state?

Reply:
It's actually the new state, since the storage handler uses the original object directly rather than create a clone. I've added a step to refresh the cache based on both the before and after states.